components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/provider/ConfigProviderImpl.java

```diff
@@ -155,6 +155,24 @@ public Object getConfigurationObject(String namespace) throws ConfigurationExcep
         return null;
     }
 
+    @Override
+    public <T> T getConfigurationObject(String namespace, Class<T> configClass) throws ConfigurationException {
+        // lazy loading deployment.yaml configuration, if it is not exists
+        loadDeploymentConfiguration(configFileReader);
+        T configObject;
```

Comment:
We can simplify this method by returning the `configObject` on the fly. That way we will not need an else statement as well. WDYT?